.travis.yml

```diff
@@ -221,325 +259,359 @@ matrix:
+        - CACHE_NAME=cronshard11
       script:
         - ./build-support/bin/travis-ci.sh -bc -i 11/20
 
-    - <<: *default_cron_test_config
+    - <<: *default_linux_test_config
       name: "Integration tests for pants (Python 2) - shard 12"
+      stage: *cron
       env:
-        - CACHE_NAME=cronshard12  # Ensure distinct caches.
+        - CACHE_NAME=cronshard12
       script:
         - ./build-support/bin/travis-ci.sh -bc -i 12/20
 
-    - <<: *default_cron_test_config
+    - <<: *default_linux_test_config
       name: "Integration tests for pants (Python 2) - shard 13"
+      stage: *cron
       env:
-        - CACHE_NAME=cronshard13  # Ensure distinct caches.
+        - CACHE_NAME=cronshard13
       script:
         - ./build-support/bin/travis-ci.sh -bc -i 13/20
 
-    - <<: *default_cron_test_config
+    - <<: *default_linux_test_config
       name: "Integration tests for pants (Python 2) - shard 14"
+      stage: *cron
       env:
-        - CACHE_NAME=cronshard14  # Ensure distinct caches.
+        - CACHE_NAME=cronshard14
       script:
         - ./build-support/bin/travis-ci.sh -bc -i 14/20
 
-    - <<: *default_cron_test_config
+    - <<: *default_linux_test_config
       name: "Integration tests for pants (Python 2) - shard 15"
+      stage: *cron
       env:
-        - CACHE_NAME=cronshard15  # Ensure distinct caches.
+        - CACHE_NAME=cronshard15
       script:
         - ./build-support/bin/travis-ci.sh -bc -i 15/20
 
-    - <<: *default_cron_test_config
+    - <<: *default_linux_test_config
       name: "Integration tests for pants (Python 2) - shard 16"
+      stage: *cron
       env:
-        - CACHE_NAME=cronshard16  # Ensure distinct caches.
+        - CACHE_NAME=cronshard16
       script:
         - ./build-support/bin/travis-ci.sh -bc -i 16/20
 
-    - <<: *default_cron_test_config
+    - <<: *default_linux_test_config
       name: "Integration tests for pants (Python 2) - shard 17"
+      stage: *cron
       env:
-        - CACHE_NAME=cronshard17  # Ensure distinct caches.
+        - CACHE_NAME=cronshard17
       script:
         - ./build-support/bin/travis-ci.sh -bc -i 17/20
 
-    - <<: *default_cron_test_config
+    - <<: *default_linux_test_config
       name: "Integration tests for pants (Python 2) - shard 18"
+      stage: *cron
       env:
-        - CACHE_NAME=cronshard18  # Ensure distinct caches.
+        - CACHE_NAME=cronshard18
       script:
         - ./build-support/bin/travis-ci.sh -bc -i 18/20
 
-    - <<: *default_cron_test_config
+    - <<: *default_linux_test_config
       name: "Integration tests for pants (Python 2) - shard 19"
+      stage: *cron
       env:
-        - CACHE_NAME=cronshard19  # Ensure distinct caches.
+        - CACHE_NAME=cronshard19
       script:
         - ./build-support/bin/travis-ci.sh -bc -i 19/20
 
 
     # Rust on linux
     - <<: *linux_with_fuse
-      name: "Rust Tests Linux"
+      name: "Linux Rust tests"
+      env:
+        - CACHE_NAME=linuxrusttests
```

Comment:
This job should use the `native_engine_cache_config` (and maybe share it with the linux bootstrap shard)

Reply:
See above. Again, is this recompiling unnecessarily? Is it better to move these tests into the bootstrap shard. Or - is it possible to propagate the bootstrapped rust build state via S3 (like we do for the pex) so that this job isn't starting from a clean rust workspace?

Reply:
These tests take a non-trivial amount of time, so we probably don't want to merge them into the bootstrap shard...

The build state we'd want to share is basically the entire cache (`~/.cache/pants/rust` + `src/rust/engine/target`), but it's probably noticeably simpler for us to just re-do the work in this shard than to do custom cross-shard caching...